async-context/README.md

```diff
@@ -0,0 +1,360 @@
+|    Host Continuation A()        |
+===================================
+```
+
+and
+
+```
+---------------------------
+| function x()            |
+---------------------------
+| function myLoggingAPI() |
+===================================
+|    continuation f2()            |
+===================================
+|    ...host code...      |
+---------------------------
+|    ...host code...      |
+===================================
+|   Host Continuation B()         |
+===================================
+```
+
+Some notes about the pictures above:
+  - We've introduced a "Root Continuation" as the bottom frame.  This  illustrates a basic assumption that all code is executing in the context of a `Continuation Invocation`.
+  - Multiple `continuations` can be on the stack at the same time, which result in mulitple `Continuation Invocations`. For any given stack frame, the "context" is the first `Continuation Invocation` below the given frame on the stack.
+
+Here, we've updated our pictures of runtime callstacks, giving  labels to the `Continuation Invocation` instances:
+
+```
+---------------------------                    -----------                            
+|   setTimeout()          |                              |
+---------------------------                              |
+|   function f1()         |                              |
+---------------------------                              \/
+|    ...host code...      |                  Continuation Invocation ci1
+---------------------------                              /\  
+|    ...host code...      |                              |
+===================================                      |
+|    Host Continuation A()        |                      |
+===================================            -----------
+```
+
+and
+
+```
+---------------------------                    -----------
+| function x()            |                              |
+---------------------------                              \/                   
+| function myLoggingAPI() |                  Continuation Invocation ci3                        
+===================================                      /\
+|    Continuation f2()            |                      |
+===================================            -----------
+|    ...host code...      |                              |
+---------------------------                              \/
+|    ...host code...      |                   Continuation Invocation ci2                   
+===================================                      /\
+|   Host Continuation B()         |                      |
+===================================            -----------
+```
+
+## Link Context
+We define the `Link Context` `link-c` of a `Continuation` `c` as a pointer from `c` to the `Continuation Invocation` instance where `c` was constructed.  In the example from above, the `continuation` `f2` has a `link context` pointing to `Continuation Invocation` `ci1`.
```

Comment:
> where `c` was constructed

I think 'where `c` was passed to a continuation point is more apt. Consider:

```js
const f1 = function() {...}
... somwhere else ...
fs.readFile(filename, f1);
```

Reply:
That was my first thought, but I kinda feel like "where constructed" is 
  - easy to implement
  - has very clear semantics
  - lets us push some anti-patterns back on the user as a "don't do that".  

Note that per current definitions in this doc, a `Link Context` is associated with the `Continuation` and not the `Continuation Invocation`.  Furthermore, a single `Continuation` can be associated with multiple `Continuation Invocations`.

Now, consider this example where we have two APIs that are `continuation points`, e.g., `fs.read()` and `stream.read()`, and `fs.readFile()` calls `stream.read()`:

```javascript
function readFile(filename, callback) {
   const stream = openFileStream(filename);
   stream.read(callback);
}
```

In the example above, if we define link context as "where a continuation is passed to a continuation point", where is the link context above established?  Whereas, if we define it as where the `continuation` is created, then it becomes very obvious.

In your example, we can push back on the user and say "don't do that". 

Happy to discuss this further.